src/routes/InventoryDetail.js

```diff
@@ -53,29 +48,25 @@ const activeApps = [
 const Inventory = () => {
     const chrome = useChrome();
     const { inventoryId } = useParams();
-    const [activeApp, setActiveApp] = useState(activeApps[0]);
-    const store = useStore();
     const { search } = useLocation();
+    const searchParams = new URLSearchParams(search);
+    const [activeApp] = useState(searchParams.get('appName') || appList[0].name);
+    const store = useStore();
     const history = useHistory();
     const dispatch = useDispatch();
-    const searchParams = new URLSearchParams(search);
     const writePermissions = useWritePermissions();
     const entityLoaded = useSelector(({ entityDetails }) => entityDetails?.loaded);
     const entity = useSelector(({ entityDetails }) => entityDetails?.entity);
     const clearNotifications = () => dispatch(actions.clearNotifications());
+
     useEffect(() => {
         chrome?.hideGlobalFilter?.(true);
         chrome.appAction('system-detail');
         clearNotifications();
-        const appName = searchParams.get('appName');
-        if (appName) {
-            dispatch(detailSelect(appName));
```

Comment:
I remember removing this action caused incorrect tab activation. I guess the new refs deal with this, right?

Reply:
Yup, this is resolved by refs